x-pack/plugins/enterprise_search/server/routes/enterprise_search/engines.ts

```diff
@@ -168,37 +169,39 @@ export function registerEnginesRoutes({ config, log, router }: RouteDependencies
       validate: { params: schema.object({ engine_name: schema.string() }) },
     },
     elasticsearchErrorHandler(log, async (context, request, response) => {
-      const engineName = decodeURIComponent(request.params.engine_name);
-      const { client } = (await context.core).elasticsearch;
+      try {
+        const engineName = decodeURIComponent(request.params.engine_name);
+        const { client } = (await context.core).elasticsearch;
 
-      const engine = await fetchEnterpriseSearch<EnterpriseSearchEngineDetails>(
-        config,
-        request,
-        `/api/engines/${engineName}`
-      );
-
-      if (!engine || (isResponseError(engine) && engine.responseStatus === 404)) {
-        return createError({
-          errorCode: ErrorCode.ENGINE_NOT_FOUND,
-          message: 'Could not find engine',
-          response,
-          statusCode: 404,
+        const engine = await client.asCurrentUser.transport.request<EnterpriseSearchEngine>({
+          method: 'GET',
+          path: `/_application/search_application/${engineName}`,
         });
-      }
-      if (isResponseError(engine)) {
-        return createError({
-          errorCode: ErrorCode.UNCAUGHT_EXCEPTION,
-          message: 'Error fetching engine',
-          response,
-          statusCode: engine.responseStatus,
+
+        const data = await fetchEngineFieldCapabilities(client, engine);
+        return response.ok({
+          body: data,
+          headers: { 'content-type': 'application/json' },
         });
+      } catch (e) {
+        if (isNotFoundException(e)) {
+          return createError({
+            errorCode: ErrorCode.ENGINE_NOT_FOUND,
+            message: 'Could not find engine',
+            response,
+            statusCode: 404,
+          });
+        }
+        if (isResponseError(e)) {
```

Comment:
Is this still needed? this was specific to `fetchEnterpriseSearch` and probably is no longer needed.